amida-tech/indaba-client
# Feature/upgrade to babel7

If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### What does this PR do?
- Upgraded all Babel packages to Babel7
- Upgraded `react-scripts` to latest version (see Background/Context below)

#### Related JIRA tickets:
- MERRY CHRISTMAS (did this just because the old stuff offended me too much -- feel free to make a ticket and assign it to me if you wanna track this)

#### How should this be manually tested?
- Test. Everything.
  - Literally everything should be regression tested. There was a lot to update here and it all needs to be rigorously tested to ensure the integrations didn't break some obscure edge-case.

#### Background/Context
- With respect to `react-scripts` -- this one is important:
  - `react-scripts` is basically the "ez launch" way to do things from React Create. It actually pulls in and manages dependent packages on its own -- ones you don't need to include in your own `package.json`. This is why I deleted some packages that are necessary like `webpack` where it may seem that I've broken the code -- `react-scripts` pulls in its own sub-modules with the versions it likes.
  - Eventually, if you want more control over your package management, you can do `react-scripts eject` which it seems Chris thought ahead of and added the `yarn eject` script. Note that you'd better be sure you actually wanna do it, since it's irreversible. But it basically adds all of its managed dependencies to your `package.json` including all of the script-config references (i.e. `jest` etc).
- I fixed some build warnings about `export 'default' (reexported as '**Actions') was not found in './actions/` since a lot of your `actions.js` files were exporting `const`s and `function`s but no `default` (which is where you'd usually export a big giant object instead of what you've got going on now). Just did this so it would stop complaining about them, but be aware that now this means you're *actually* exporting all your actions externally. I'm leaving it up to you on how you wanna handle that, or leave it be, etc.

#### Screenshots (if appropriate):
